sdk/appconfiguration/azappconfiguration/client_responses.go

```diff
@@ -0,0 +1,102 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azappconfiguration
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+
+	"sdk/appconfiguration/sdk/appconfiguration/azappconfiguration/internal/generated"
+)
+
+type ConfigurationSettingResult struct {
+	ConfigurationSetting
+	ETag      *azcore.ETag
+	SyncToken *string
+}
+
+type configurationSettingResponse struct {
```

Comment:
We should be exporting the response type

Reply:
It is private, but is made public by
```
type AddConfigurationSettingResponse configurationSettingResponse
type DeleteConfigurationSettingResponse configurationSettingResponse
type SetConfigurationSettingResponse configurationSettingResponse
type SetReadOnlySettingResponse configurationSettingResponse // soon
```

Reply:
My thought process is that we have specialized types for a response for each API (so that in the future if only one of them adds an extra property, we modify just one, and don't make a breaking change. So, users see we have "separate" types, but internally we don't copy&paste the same struct four times.

Reply:
That's a fine reasoning, I don't know if that's the most idiomatic way of doing it. @jhendrixMSFT what are your thoughts?

Reply:
Also, how does this show up on intellisense?

Reply:
A view from another package:
<img width="566" alt="image" src="https://user-images.githubusercontent.com/41349689/154606929-bd886a3f-d806-4538-9d6f-2374701fbaab.png">

<img width="483" alt="image" src="https://user-images.githubusercontent.com/41349689/154607133-46515514-8888-463d-a301-5e0d6093c1eb.png">

<img width="503" alt="image" src="https://user-images.githubusercontent.com/41349689/154607360-bb408b0f-e56d-431c-86c3-cebeb153d532.png">

<img width="623" alt="image" src="https://user-images.githubusercontent.com/41349689/154607485-5c0fae2a-1e1f-47bb-9afc-f359ce34bd86.png">

Reply:
I think the screenshots look good.